src/main/java/duke/task/TaskManager.java

```diff
@@ -0,0 +1,276 @@
+     * @param type           Type of task given by user input
+     */
+    public void addTask(String[] userInputArray, TaskType type) {
+
+        // Check for empty description and return if no description
+        try {
+            boolean emptyDescription = userInputArray[1].isBlank();
+            if (emptyDescription) {
+                Duke.printMessage(EMPTY_DESCRIPTION_ERROR_MESSAGE);
+                return;
+            }
+
+            // Create tasks
+            Task task = null;
+            switch (type) {
+            case TODO:
+                task = createTask(userInputArray[1], CHAR_TYPE_TODO);
+                break;
+            case EVENT:
+                task = createTask(userInputArray[1], CHAR_TYPE_EVENT);
+                break;
+            case DEADLINE:
+                task = createTask(userInputArray[1], CHAR_TYPE_DEADLINE);
+                break;
+            }
+
+            // As long as task is valid, add to task list and inform user
+            if (task != null) {
+                tasksList.add(task);
+                Duke.printMessage(String.format(ADD_TASK_MESSAGE, task, tasksList.size()));
+            }
+
+        } catch (IndexOutOfBoundsException e) {
+            Duke.printMessage(EMPTY_DESCRIPTION_ERROR_MESSAGE);
+        } catch (InvalidParameterException e) {
+            Duke.printMessage(e.getMessage());
+        }
+
+    }
+
+    /**
+     * Prints the entire list of tasks currently in tasks list. Will let user know if there is no task in tasks list.
+     */
+    public void listTask() {
+        int tasksListSize = tasksList.size();
+        String message = "";
+        switch (tasksListSize) {
+        case 0:
+            message = "Oh! You have no tasks left!";
+            break;
+        default:
+            message = "Total of " + tasksListSize + " task(s)\n";
+            int counter = 1;
+            for (Task task : tasksList) {
+                message += String.format("%d.%s\n", counter, task.toString());
+                counter++;
+            }
+            break;
+        }
+        Duke.printMessage(message);
+    }
+
+    /**
+     * Check if task exist
+     *
+     * @param taskNumber Task Number given by user
+     * @return Corresponding task index fpr task number
+     * @throws InvalidParameterException If task index does not exist in the current task list
+     */
+    public int checkTaskExist(int taskNumber) throws InvalidParameterException {
+        boolean isNotWithinSizeLimit = taskNumber < 1 || taskNumber > tasksList.size();
+        if (isNotWithinSizeLimit) {
+            throw new InvalidParameterException(TASK_DOES_NOT_EXIST_MESSAGE);
+        }
+        return taskNumber - 1;
+    }
+
+    /**
+     * Set the task to be completed by marking it done.
+     *
+     * @param taskNumber task number given by user
+     */
+    public void completeTask(int taskNumber) {
+        try {
+            int taskNumberIndex = checkTaskExist(taskNumber);
```

Comment:
The method name `checkTaskExist` makes it sound like it will return a boolean but instead, it returns the index. You can consider renaming the method to `getIndexOfTask` instead to improve code readability.